controlplane/internal/controllers/rke2controlplane_controller.go

```diff
@@ -742,14 +743,23 @@ func (r *RKE2ControlPlaneReconciler) upgradeControlPlane(
 		return ctrl.Result{}, err
 	}
 
-	status := workloadCluster.ClusterStatus()
+	switch rcp.Spec.RolloutStrategy.Type {
+	case controlplanev1.RollingUpdateStrategyType:
+		// RolloutStrategy is currently defaulted and validated to be RollingUpdate
+		// We can ignore MaxUnavailable because we are enforcing health checks before we get here.
```

Comment:
Q: what is the use of `MaxUnavailable` in this context? AFAICT it is only MaxSurge we are introducing in this PR

Reply:
There is no use, removed the comment.